### PR TITLE
fix: allow also " inside of an embed

### LINKED ISSUE #1005: Embed files does not work
## Bug Report
By using the last release of docsify (from CDN https://cdn.jsdelivr.net/npm/docsify@4), the feature Embed files does not work.

#### Steps to reproduce

From the code below :
```
[filename](./test.md ":include")
```

#### What is current behaviour
The render will be a link name filename.

#### What is the expected behaviour
The test.md content should be rendered directly (embedded).

#### Other relevant information

- [ ] Bug does still occur when all/other plugins are disabled?

- Your OS: Windows
- Node.js version: v10.15.3
- npm/yarn version: npm 6.4.1
- Browser version: Chrome
- Docsify version: 4.4.0
- Docsify plugins: tried without any plugins, but I have the same issue

### LINKED ISSUE #1555: ':include' not working in Docsify 4.12.1
> image
>
image

> 
> Details:
> 
> * OS: macOS 11.2.3
> * Node.js version: 15.14.0
> * npm version: 7.8.0
> * Browser version: chrome 89.0.4389.114
> * Docsify version: 4.12.1
>

### LINKED ISSUE #1587: ":include" in double quotes doesn't recognized
#### Steps to reproduce

Try to embed external file using for example this link:
```
[readme](https://raw.githubusercontent.com/softvisio/core/main/README.md ":include")
```

#### What is current behaviour

Currently file will not be embedded, because `:include` tag is placed in double quotes.
This is the problem, because `prettier` markdown processor replaces single quotes with double and double quotes are legal for the link title.

#### What is the expected behaviour

I am expecting, that file will be embedded ti the parent document.

#### Other relevant information

- [ x] Bug does still occur when all/other plugins are disabled?

- Your OS: 
- Node.js version: 
- npm/yarn version: 
- Browser version: 
- Docsify version: 4.12.1
- Docsify plugins: 

#### Please create a reproducible sandbox 

[Edit 307qqv236](https://codesandbox.io/s/307qqv236)

### LINKED ISSUE #1597: embed quotes bug
#### Steps to reproduce
1. Create empty docsify project
2. Add following to one markdown file
  ```markdown
  [gist: script.js](https://gist.githubusercontent.com/anikethsaha/f88893bb563bb7229d6e575db53a8c15/raw/script.js ":include :type=code")
  ```
3. See not renderd embed 

#### What is current behaviour

Embedded gist will not be renderd.

#### What is the expected behaviour

Embedded gist should be renderd.

#### Other relevant information

This is not working because the type of quotes are not supported. The default prettier configuration changes the `''` to  `""`. And if you have auto format toggled on, you're not be able to save the file working.

- [x] Bug does still occur when all/other plugins are disabled?

- Your OS: any os
- Node.js version: any nodejs version
- npm/yarn version: any yarn/npm version
- Browser version: any browser and version
- Docsify version: 4.12.1
- Docsify plugins: none

#### Please create a reproducible sandbox 

[Edit docsify-template-forked-zlwql](https://codesandbox.io/s/docsify-template-forked-zlwql)

### PR DESCRIPTION
**Pull request by luwol03**
## **Summary**

I fixed a bug where `":include"` is not handled like `':include'`. This is not working because the type of quotes are not supported. The default prettier configuration changes the `''` to `""`. And if you have auto format toggled on, you're not be able to save the file with the right quotes.

## **What kind of change does this PR introduce?**

I added added the `"` also the the replace regex

```js
str = str
      .replace(/^('|")/, '')
      .replace(/('|")$/, '')
```

## **For any code change,**

- [x] Related documentation has been updated if needed
- [ ] Related tests have been updated or tests have been added (I cant find some tests for that already.)

## **Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

## **Related issue, if any:**

closes #1597 
closes #1587 
closes #1555 
closes #1005 

## **Tested in the following browsers:**

- [x] Chrome
- [x] Firefox
- [ ] Safari
- [ ] Edge
- [ ] IE

### CODE REVIEW

**Review by sy-records — COMMENTED** [review]
Can you add a test for this?

**Comment by Koooooo-7** [comment]
@sy-records 
As we all know, this func `getAndRemoveConfig ` has been used for much places.
It seems we need separate it for different functions in case of unpredicted issues.
WDTY?

**Review by Koooooo-7 — COMMENTED** [review]
@sy-records
As we all know, this func getAndRemoveConfig has been used for much places.
It seems we need separate it for different functions in case of unpredicted issues.
WDTY?

**Comment by sy-records** [comment]
Yes, you're right.

**Comment by trusktr** [comment]
> It seems we need separate it for different functions in case of unpredicted issues.

@Koooooo-7 What do you mean?

**Comment by Koooooo-7** [comment]
> > It seems we need separate it for different functions in case of unpredicted issues.
> 
> @Koooooo-7 What do you mean?

This func r working in multi place to get configs. such as titles ,  sidebar headings.
If we add " for embed files,  it might cause other problems in other places.
so, we may need separate this func , for sidebar, for titles and for embed files.

**Comment by trusktr** [comment]
@Koooooo-7 I see what you're saying, although it seems very unlikely. My guess would be that there is no one relying on he non-working double quotes.

It seems multiple people have hit this issue, and we closed the issues. It was in fact working before, then it broke:

- https://github.com/docsifyjs/docsify/issues/1587
- https://github.com/docsifyjs/docsify/issues/1555
- https://github.com/docsifyjs/docsify/issues/1005

**Comment by Koooooo-7** [comment]
@trusktr  I checked this func history, it seems never support the `"` before, the docs either.
You could check [this func](https://github.com/docsifyjs/docsify/pull/708/files) since `Dec 5, 2018` and earlier.
If we support it right now, it would remove the `"` in other places also. 
So, I prefer to separate this func only for different places.

**Comment by luwol03** [comment]
> Can you add a test for this?

sure, will try that.

> > > It seems we need separate it for different functions in case of unpredicted issues.
> > 
> > 
> > @Koooooo-7 What do you mean?
> 
> This func r working in multi place to get configs. such as titles , sidebar headings.
> If we add " for embed files, it might cause other problems in other places.
> so, we may need separate this func , for sidebar, for titles and for embed files.

I dont think so. I would also like to use the double ones inside of headlines, or other stuff. Everywhere I can use this style.

And another question, do you prefere an clean commit history, so should I rebase merge upstream changes?

**Comment by Koooooo-7** [comment]
@luwol03  of cuz we can support this feat for those configs. instead, how about those titles/other stuff without configs but with " ?

**Comment by luwol03** [comment]
> @luwol03 of cuz we can support this feat for those configs. instead, how about those titles/other stuff without configs but with " ?
> e.g a title `this is "my title"`.

Then we should improve the function, so that it only replaces the quotes, if they are followed by a `:xxxx` and there is also a closing one.

**Comment by Koooooo-7** [comment]
> > @luwol03 of cuz we can support this feat for those configs. instead, how about those titles/other stuff without configs but with " ?
> > e.g a title `this is "my title"`.
> 
> Then we should improve the function, so that it only replaces the quotes, if they are followed by a `:xxxx` and there is also a closing one.

agreed.
If possible, glad to see a good solution for this func/regx to solve those issues above.

**Review by Koooooo-7 — APPROVED** [review]

**Comment by luwol03** [comment]
Is this normal, that your e2e tests failed?

**Comment by Koooooo-7** [comment]
> Is this normal, that your e2e tests failed?

It seems not this changes broken, it's okay.
I would retry it.

**Comment by trusktr** [comment]
> If we support it right now, it would remove the `"` in other places also.

@Koooooo-7 can you point them out?

**Comment by trusktr** [comment]
> Is this normal, that your e2e tests failed?

Sometimes it fails for some reason despite that it shouldn't. Needs a fix.

**Comment by luwol03** [comment]
Before merge, I will try that:

> Then we should improve the function, so that it only replaces the quotes, if they are followed by a :xxxx and there is also a closing one.

**Comment by trusktr** [comment]
> Before merge, I will try that:
> 
> > Then we should improve the function, so that it only replaces the quotes, if they are followed by a :xxxx and there is also a closing one.

Is anyone using it without the `:`?

**Comment by trusktr** [comment]
Here is an example without the colon: https://github.com/docsifyjs/docsify/issues/1587#issuecomment-854293984

File: test/unit/render-util.test.js
Comment (trusktr) [inline test/unit/render-util.test.js:61]: I wonder what the behavior should actually be. Seems funky that it leaves things behind. Seems like it should handle everything.

**Review by trusktr — APPROVED** [review]

**Comment by trusktr** [comment]
@jhildenbiddle One test or other keeps randomly failing.

**Comment by trusktr** [comment]
Investigating the failures. Let's not merge until we know it isn't the change from here. Seems unlikely that it is.

**Review by trusktr — CHANGES_REQUESTED** [review]
Looks good. I'm not requesting any changes, just blocking from merging to ensure the test is just flaky (that's what it seems like).

**Review by trusktr — APPROVED** [review]

**Comment by trusktr** [comment]
> improve the function, so that it only replaces the quotes, if they are followed by a :xxxx and there is also a closing one.

I don't think we should do that, it could break this: `"this-is-currently-ignored :but=this-currently-works"`. If we do, it can be a new PR.

Thanks @luwol03! Merging, since tests pass except one at random, which seems unrelated.

**Comment by luwol03** [comment]
Oh, A few seconds ago, I pushed my changes, before I relized, that you already merged this changes :joy:

**Comment by sy-records** [comment]
@luwol03 You can open a new PR

**Comment by luwol03** [comment]
> @luwol03 You can open a new PR
@sy-records done - see #1603

**Pull request merged**
